### ISSUE TITLE
MNT: End-of-life timeline + archiving repo?

### ISSUE THREAD

**Issue by pllim** [astropy/astropy-helpers#510]
Now that the core no longer uses it, maybe there needs to be a roadmap to put an end-of-life to this thing and then archive the repo as read only.

**Comment by bsipocz** [comment]
LTS is still on the helpers, as well as plenty of coordinated packages, so this has to go on for another 1.5 years at least.

**Comment by pllim** [comment]
Jan 2022?

**Comment by Cadair** [comment]
We should still document that in the README?

**Comment by bsipocz** [comment]
I don't think we should yet, but once most of the packages made the transition. Core is one thing, but yet a lot to do within the closer ecosystem. I would suggest don't say/plan much about the archiving until at least the 8 coordinated packages made the transition.

**Comment by Cadair** [comment]
A year and half + is a long time, and I think the sooner we put something somewhere visible saying the plan is to archive this the better.

If we aren't yet completely sure on the time line we could say something like "after the end of life of Astropy 4.0". I think having that there will drive it home to people that this is *eventually* going away.

Being transparent about the plan here is the most important thing we can do to shorten the amount of time we need to maintain this (which I appreciate we can't do until most people stop using it).

**Comment by bsipocz** [comment]
> If we aren't yet completely sure on the time line we could say something like "after the end of life of Astropy 4.0". I think having that there will drive it home to people that this is eventually going away.

I'm OK with a phrasing like this, the only thing I feel is premature to put an exact date, like Jan 2022, down.

**Comment by astrojuanlu** [comment]
What do folks think about making a release that includes #504 (fix for #501)? Got this while trying to install astroquery (-> pyvo -> https://github.com/astropy/pyvo/issues/228)

<details>
Longer text I was about to write as part of a new issue.

I got an installation error with `astropy-helpers` that was already solved in #504 (see #501 for original report). It would be nice to have a new release that fixes this.

Even though `astropy-helpers` is deprecated, as highlighted by #510 it's still used by many coordinated packages. In fact, the reason why I got this error is because astroquery depends on pyvo (besides astropy-helpers itself):

https://github.com/astropy/astroquery/blob/98e9537ef1e86a85e775b42c0522ed7c87aa6d52/setup.cfg#L136

which, in turn, depends on astropy-helpers.

There are a few commits since the last release https://github.com/astropy/astropy-helpers/compare/v4.0.1...master therefore I understand that this might be potentially disruptive. Since #501 is, if anything, a minor annoyance that can be easily solved, I'd like to ask whether you think it's worth doing.

Of course, the actual fix for the problem is to migrate such packages off astropy-helpers :) But it's a large infrastructure change that might need some effort. The other alternative is to publish wheels for all these packages.
</details>

**Comment by pllim** [comment]
Thanks for bringing this to our attention! @bsipocz might have fixed the `astroquery` issue very recently. Let's see what she says first.

**Comment by bsipocz** [comment]
No, it hasn't been fixed directly, but also we never run into the issue, I suppose `wheel` is getting installed indirectly as dependency for upstream by the time we install pyvo. 
(For astroquery I've built a wheel, too, getting one for pyvo use probably the quickest).

**Comment by astrofrog** [comment]
Uploading a PyVO wheel is the minimal effort fix so I'm not enclined to spend much time on an astropy helpers release because anyway this would require updating the sub module in PyVO and doing a new release 🤷‍♂️

**Comment by astrojuanlu** [comment]
Thanks all, indeed @astrofrog uploading a PyVO wheel sounds like the way to go.

**Comment by pllim** [comment]
cc @cbanek 🙏

**Comment by bsipocz** [comment]
I think it's fair to archive the repo.
 (We are still using it as nothing really is broken in it that affects us and therefore switching has the lowest, if not negative priority. If anyone has extra energy to do infrastructure, then it should be spent on fixing caching 😆)

**Comment by bsipocz** [comment]
Yes, I saw it and thus I'm totally OK with archiving the repo. We'll keep using it though and will switch instead of fixing it if something comes up.

**Comment by pllim** [comment]
Okay. I added a note to README about archiving at [3b45ed3](https://github.com/astropy/astropy-helpers/commit/3b45ed3191ceb45c574db304ec0f33282d2e4a98) . I am closing this issue so I can archive. Thank you!
